File: 263-UglyNumber/263-UglyNumber.cpp
// Last updated: 13/07/2026, 22:21:59
class Solution {
public:
    bool isUgly(int n) {
        if (n < 1) return false;
        if (n == 1) return true;
        n = abs(n);
        while (n % 2 == 0) n = n / 2;
        while (n % 3 == 0) n = n / 3;
        while (n % 5 == 0) n = n / 5;
        return n == 1;
    }
};

File: 263-UglyNumber/263-UglyNumber.py
# Last updated: 13/07/2026, 22:22:11
class Solution:
    def isUgly(self, n: int) -> bool:
        primes_allowed = [2,3,5]
        cur_index = 0
        while (cur_index < 3 and n > 1):
            if not(n % primes_allowed[cur_index]):
                n = n // primes_allowed[cur_index]
            else:
                cur_index += 1
        
        return n == 1